src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -6,11 +6,20 @@
 #include <string.h>
 #include <vector>
 
-#include "config.h"
+#include "icushim.h"
 #include "locale.hpp"
 #include "holders.h"
 #include "errors.h"
 
+#ifndef FEATURE_FIXED_ICU_VERSION
+// The System.Globalization.Native.so is built in mode when it can use any supported version
+// of ICU installed on the target machine and not just the one it was built with. So we can 
+// only use stable APIs and the UDAT_STANDALONE_SHORTER_WEEKDAYS doesn't belong to those.
+// We also want to support ICU versions where the UDAT_STANDALONE_SHORTER_WEEKDAYS was not 
+// defined yet.
+#undef HAVE_UDAT_STANDALONE_SHORTER_WEEKDAYS
```

Comment:
I would prefer to keep the configure step agnostic of what we are building and just providing information of what is available during the build.
Now this HAVE_SET_MAX_VARIABLE is different from the HAVE_UDAT_STANDALONE_SHORTER_WEEKDAYS. If you have ICU that already has the ucol_setMaxVariable at the build time and you'd try to use the old ucol_setVariableTop, the compilation will fail with warning on the ucol_setVariableTop being deprecated. 
It is actually interesting what does it really mean for a stable api like this to be deprecated. Based on what they say on the stable APIs, it should still exist in the future versions of the library, so I assume it won't be removed in the future.
I am actually starting to think that I should change the minimum major version number of ICU that we support to be the version of ICU we are building against. That would mean that when we want to build a version of System.Globalization.Native.so that would work on a set of supported distros, we would build it on a distro with the oldest dependencies from all the distros from the set. That's what we actually need to do for glibc as well to make sure we don't reference a newer version of a function from glibc - from time to time, new versions of the same functions are added to glibc while the older ones remain.
If we did it this way, then we would keep the HAVE_SET_MAX_VARIABLE usage as I have it now.
Does it make sense?